task/sql-tasks.js

```diff
@@ -280,7 +475,26 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+        SELECT DISTINCT
+            Customers.CompanyName,
+            Products.ProductName,
+            OrderDetails.UnitPrice AS "PricePerItem"
+        FROM Customers 
+        JOIN Orders ON Customers.CustomerID = Orders.CustomerID
+        JOIN OrderDetails ON Orders.OrderID = OrderDetails.OrderID
+        JOIN Products ON OrderDetails.ProductID = Products.ProductID
+        WHERE OrderDetails.UnitPrice = 
+            (SELECT
+                MAX(orderdetails.UnitPrice)
+            FROM Customers customers1
+            JOIN Orders orders ON orders.CustomerID = customers1.CustomerID
+            JOIN OrderDetails orderdetails ON orderdetails.OrderID = orders.OrderID
+            WHERE Customers.CompanyName = customers1.CompanyName

```

Comment:
company name doesn't uniquely identify customer. also this column doesn't have an index so it's also slow.

Reply:
also try to group by customer id and join subquery on price and customer id